### ISSUE TITLE
Integrate trueno-viz for training visualization

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#124]
## Summary

Add optional integration with [trueno-viz](https://crates.io/crates/trueno-viz) (v0.1.5) for training visualization capabilities.

## Motivation

Currently aprender has no visualization capabilities. trueno-viz provides SIMD-accelerated visualization that could enhance the developer experience for:
- Training curve visualization (loss over epochs)
- Model inspection plots
- Terminal-based dashboards for CLI users

## Proposed Integration

```toml
[features]
viz = ["trueno-viz"]
```

### Features to integrate

| Feature | Description | Use Case |
|---------|-------------|----------|
| ScatterPlot | 2D scatter visualization | Data exploration, cluster visualization |
| ASCII output | Terminal rendering | CLI model inspection |
| Framebuffer | Pixel-level control | Custom visualizations |

### Example API

```rust
use aprender::viz::TrainingPlot;

let plot = TrainingPlot::new("Loss over Epochs");
for epoch in 0..100 {
    let loss = train_epoch(&model, &data);
    plot.add_point(epoch, loss);
}
plot.render_ascii();  // Terminal output
```

## Priority

LOW - Nice-to-have enhancement, not blocking any functionality.

## References

- trueno-viz: https://crates.io/crates/trueno-viz
- Spec: `docs/specifications/include-latest-trueno-features.md`

**Comment by noahgift** [comment]
Already integrated - `trueno-viz = "0.1.16"` is in Cargo.toml with `visualization` feature flag.
